plugin/vcs/gitlab/gitlab.go

```diff
@@ -214,6 +214,28 @@ func (p *Provider) APIURL(instanceURL string) string {
 	return fmt.Sprintf("%s/%s", instanceURL, apiPath)
 }
 
+// gitlab oauth response
+type oauthResponse struct {
+	AccessToken      string `json:"access_token" `
+	RefreshToken     string `json:"refresh_token"`
+	ExpiresIn        int64  `json:"expires_in"`
+	CreatedAt        int64  `json:"created_at"`
+	ExpiresTs        int64  `json:"expires_ts"`
```

Comment:
I guess we don't need this, expiresTs can be calculated in toVCSOAuthToken

Reply:
Can wait for the refactoring to be done together

Reply:
move to toVCSOAuthToken() done